test/functional/system/rounds.js

```diff
@@ -0,0 +1,1538 @@
+								let genesisAccount;
+								let genesisAccountTransaction;
+
+								before(done => {
+									genesisAccountTransaction =
+										library.genesisblock.block.transactions[0];
+									genesisAccount = _.find(accounts, {
+										address: genesisAddress,
+									});
+									done();
+								});
+
+								it('should exists', () => {
+									return expect(genesisAccount).to.be.an('object');
+								});
+
+								it('balance should be negative', () => {
+									return expect(Number(genesisAccount.balance)).to.be.below(0);
+								});
+
+								it('fields address, balance, publicKey should match genesis block transaction', done => {
+									expect(genesisAccount.address).to.equal(
+										genesisAccountTransaction.senderId
+									);
+
+									// Sum all outgoing transactions from genesis account
+									const balance = _.reduce(
+										library.genesisblock.block.transactions,
+										(balance, acc) => {
+											if (acc.senderId === genesisAccount.address) {
+												return new Bignum(balance).minus(acc.amount).toString();
+											}
+											return balance;
+										},
+										'0'
+									);
+
+									expect(genesisAccount.balance).to.be.equal(balance);
+									expect(genesisAccount.u_balance).to.be.equal(balance);
+									expect(genesisAccount.publicKey.toString('hex')).to.equal(
+										genesisAccountTransaction.senderPublicKey
+									);
+									done();
+								});
+							});
+
+							describe('all accounts', () => {
+								it('balances should be valid against blockchain balances', () => {
+									// Perform validation of accounts balances against blockchain
+									return expect(
+										Queries.validateAccountsBalances()
+									).to.eventually.be.an('array').that.is.empty;
+								});
+							});
+						});
+					});
+				});
+			});
+		});
+
+		describe('modules.delegates', () => {
+			describe('__private.delegatesList', () => {
+				let delegatesList;
+
+				before(() => {
+					return generateDelegateListPromise(1, null).then(_delegatesList => {
+						delegatesList = _delegatesList;
+					});
+				});
+
+				it('should be an array', () => {
+					return expect(delegatesList).to.be.an('array');
+				});
+
+				it('should have a length of 101', () => {
+					return expect(delegatesList.length).to.equal(101);
+				});
+
+				it('should contain public keys of all 101 genesis delegates', done => {
+					_.each(delegatesList, pk => {
+						// Search for that pk in genesis block
+						var found = _.find(library.genesisblock.block.transactions, {
+							senderPublicKey: pk,
+						});
+						expect(found).to.be.an('object');
+					});
+					done();
+				});
+
+				it('should be equal to one generated with Lisk-Core 0.9.3', () => {
+					return expect(delegatesList).to.deep.equal(
+						roundsFixtures.delegatesOrderAfterGenesisBlock
+					);
+				});
+			});
+
+			describe('__private.loadDelegates', () => {
```

Comment:
why do we have these tests in rounds functional tests?

Reply:
Because rounds tests depend on this, before start actual tests would be nice to be sure that delegates were loaded correctly.

Reply:
Moved those, together with genesis block tests, to application functional system tests.